lib/Utils/dir_list_ci.cpp

```diff
@@ -55,59 +103,144 @@ std::string DirListCI::resolveFileCase(const std::string &in_name)
+        auto found = m_fileMap.find(uppercase_string);
+        if(found != m_fileMap.end())
+            return found->second;
     }
 
-    return name;
+    return "";
+}
+
+std::string DirListCI::resolveFileCase(const std::string &in_name)
+{
+#if 0
+    return in_name; // no need on Windows
+#else
+    if(in_name.empty())
+        return in_name;
+
+    std::string found = resolveFileCaseExists(in_name);
+
+    // if not found, overwrite with the replace-slash version of the name
+    // and return
+    if(found.empty())
+        replaceSlashes(found, in_name);
+
+    return found;
+#endif
+}
+
+std::string DirListCI::resolveFileCaseAbs(const std::string &in_name)
+{
+#if 0
+    return m_curDir + "/" + in_name; // no need on Windows
+#else
+    if(in_name.empty())
+        return in_name;
+
+    std::string found = resolveFileCaseExists(in_name);
+
+    // if not found, overwrite with the replace-slash version of the name
+    // and return
+    if(found.empty())
+        replaceSlashes(found, in_name);
+
+    return m_curDir + "/" + found;
 #endif
 }
 
+std::string DirListCI::resolveFileCaseExistsAbs(const std::string &in_name)
+{
+    if(in_name.empty())
+        return in_name;
+
+    std::string found = resolveFileCaseExists(in_name);
+
+    // if not found, overwrite with the replace-slash version of the name
+    // and return
+    if(found.empty())
+        return found;
+
+    return m_curDir + "/" + found;
+}
+
 std::string DirListCI::resolveDirCase(const std::string &name)
 {
-#if 0 //def _WIN32
+#if 0
     return name; // no need on Windows
 #else
     if(name.empty())
         return name;
 
-    for(std::string &c : m_dirList)
-    {
-        if(SDL_strcasecmp(c.c_str(), name.c_str()) == 0)
-            return c;
-    }
+    std::string uppercase_string;
+    uppercase_string.resize(name.length());
+    std::transform(name.begin(), name.end(), uppercase_string.begin(),
+        [](unsigned char c){ return std::toupper(c); });
 
-    return name;
+    auto found = m_dirMap.find(uppercase_string);
+    if(found == m_dirMap.end())
+        return name;
+
+    return found->second;
 #endif
 }
 
 void DirListCI::rescan()
 {
-    m_fileList.clear();
-    m_dirList.clear();
+    m_fileMap.clear();
+    m_dirMap.clear();
     if(m_curDir.empty())
         return;
 
     DirMan d(m_curDir);
-    d.getListOfFiles(m_fileList);
-    d.getListOfFolders(m_dirList);
+    std::vector<std::string> fileList;
+    std::vector<std::string> dirList;
+    d.getListOfFiles(fileList);
+    d.getListOfFolders(dirList);
+
+    std::string uppercase_string;
+
+    for(std::string& file : fileList)
+    {
+        uppercase_string.resize(file.length());
+        std::transform(file.begin(), file.end(), uppercase_string.begin(),
+            [](unsigned char c){ return std::toupper(c); });
+        m_fileMap.insert(std::make_pair(uppercase_string, file));
+    }
+    for(std::string& dir : dirList)
```

Comment:
I'd added the empty line here, otherwise, this looks clobbish

Reply:
Yes, that is a great suggestion!